discovery-provider/integration_tests/queries/test_get_prev_track_entries.py

```diff
@@ -0,0 +1,51 @@
+from integration_tests.utils import populate_mock_db
+from src.queries.notifications import get_prev_track_entries
+from src.utils.db_session import get_db
+
+def test_get_prev_track_entries(app):
+    """Tests that prev track entries query properly returns previous tracks"""
+    with app.app_context():
+        db = get_db()
+
+    test_entities = {
+        "tracks": [
+            { "track_id": 1, "is_current": False, "is_unlisted": True, "remix_of": None }, # Block 0
+            { "track_id": 2, "is_current": True, "is_unlisted": False, "remix_of": None }, # Block 1
+            { "track_id": 3, "is_current": False, "is_unlisted": False, "remix_of": None }, # Block 2
+            { "track_id": 4, "is_current": False, "is_unlisted": False, "remix_of": None }, # Block 3
+            { "track_id": 5, "is_current": False, "is_unlisted": False, "remix_of": None }, # Block 4
+            { "track_id": 6, "is_current": False, "is_unlisted": True, "remix_of": None }, # Block 5
+            { "track_id": 1, "is_current": True, "is_unlisted": False, "remix_of": None }, # Block 6
+            { "track_id": 3, "is_current": False, "is_unlisted": False, "remix_of": None }, # Block 7
+            { "track_id": 6, "is_current": True, "is_unlisted": False, "remix_of": None }, # Block 8
+            { "track_id": 4, "is_current": True, "is_unlisted": False, "remix_of": None }, # Block 9
+            { "track_id": 3, "is_current": True, "is_unlisted": False, "remix_of": None }, # Block 10
+            { "track_id": 5, "is_current": False, "is_unlisted": False, "remix_of": None }, # Block 11
+            { "track_id": 5, "is_current": True, "is_unlisted": False, "remix_of": None }, # Block 12
+        ],
+    }
+
+    populate_mock_db(db, test_entities)
+
+    with db.scoped_session() as session:
+        entries = [
+            { "track_id": 6, "blocknumber": 8 },
```

Comment:
IIUC, we only are testing things here where the query for "blocknumber" is that latest blocknumber in that db for the track. We should add a test to make sure we're taking the "slice" / "window" properly ,e.g. add a row here

`{ "track_id": 5, "blocknumber": 11 }`

In the result, we would expect id 5, blocknumber 4 to come back.

Can we do that?